### DISCUSSION TITLE
Feature Request: Limit the maximum allowed resources leased on the provider.

### DISCUSSION THREAD

**Discussion by shimpa1** [akash-network/community#404] [Ideas]
**The problem:**
We have a huge influx of CPU miners monopolizing all CPU resources. While this is great for business, it creates unintended negative consequences.

1. If all the CPU resources are leased and nothing is left (as is the case at the moment of writing this), it is not possible to update existing deployments. Basically the scheduler will not destroy the existing deployment until a new instance has been created. The problem is that there's not enough resources to start a new deployment without destroying the old one first. This leads to unintended lease closures and is bad for long-term clients and reputation.
2. If the provider is offering other resources to lease such as persistent storage, IP addresses, GPUs etc and every CPU unit is already leased, every other resource does not get the chance to get leased and is being wasted. An example is Europlots provider which has 3 available GPUs and 9TB of persistent storage available which didn't get leased once since the CPU mining frenzy started.

**Suggested bandaging:**
- Create a configurable rule that would leave a bit of other resources reserved so the deployments can get updated without issues.
- Create a configurable rule that would count the number of unused GPUs and reserve a bit of resources so that the offers don't get passed on because of lack of a single resource. An example would be to leave 32GB of RAM and 8 or 16 VCPU per unleased GPU free. Again make it configurable so that every provider admin can choose how they want to do it.

thank you!

Shimpa | Europlots

**Comment by SGC41** [comment]
think most of this could be done with some slightly advanced changes to the price_script_generic.sh

sounds like a great idea, i have certainly also seen the same issues on DCnorse provider, but pretty early on i raised my CPU prices a lot,
to try and free up some CPU for GPU leases.

**Reply by piber-dev** [reply]
agreed with how the implementation could look like. providers can also create their own strategies in their price scripts to only bid on deployments with certain ratios between two resources. here's an example from [this guide from the recent hackathon](https://github.com/figurestudios/designing-price-script-guide),

```
# example: server with 128 threads and 128 GB of ephemeral storage where we want the threads to fill up first, or low disk usage on our leases.

minimum_resource_ratio=1 # 128/128 = 1 thread/GB ratio
if [[ ($cpu_requested / $ephemeral_storage_requested) -lt $minimum_resource_ratio ]]; then
  exit 1
fi
```
similarly something could be created to do the same but with reserving resources for updating deployments or for GPUs like @shimpa1 requested. not sure how you read your currently rented resources vs. your total resources the best way in this price script, but it's possible.

**Reply by troian** [reply]
@shimpa1 as above mentioned, the bid script would serve as best option for now
